grails-app/views/inventoryItem/_productDetails.gsp

```diff
@@ -424,6 +415,21 @@
 	function closeDialog(dialogId, imgId) {
 		$(dialogId).dialog('close');
 	}
+
+    $(window).load(function(){
+      $.ajax({
+        dataType: "json",
+        timeout: 120000,
```

Comment:
Why 2 min timeout?

Reply:
@awalkowiak I think @DominikaHatala was just copying the timeout from somewhere else. It probably only needs to be 15-20 seconds for now. Or better yet, I've added a configuration property to Config.groovy that we can use here. It might be too high but the ajax request is a temporary solution for now.
```
openboxes.ajaxRequest.timeout = 120000
```

So you could do something like this
```
var timeout = ${grailsApplication.config.openboxes.ajaxRequest.timeout?:0}
$.ajax({
    dataType: "json",
    timeout: timeout,
    ...
});
```
For what it's worth, this is fine for now. If we end up needing to do the same thing (pull data from the server) somewhere else it might be good to make this solution more DRY so we can reuse it. We could even build a taglib where we'd specify the URL, DOM ID and attribute key and the value would magically be displayed.


_productDetails.gsp

    <g:remoteRequest url="${request.contextPath}/json/getForecastingData/${productInstance?.id}"
        id="#onHandMonths" 
        value="onHandMonths" 
        dataType="decimal"
        template="${0} months"/>
    
RemoteRequestTaglib.groovy

    def remoteRequest = { 
        def defaultTimeout = grailsApplication.config.openboxes.ajaxRequest.timeout?:0
        attrs.timeout = attrs.timeout?:defaultTimeout
        render(template: "remoteRequest", model: [attrs:attrs])
    }



_remoteRequest.gsp

    <div id="${attrs.id}"></div>

    <g:javascript>
        // FIXME (pseudocode) need to populate this with the ids from attrs 
        // I hate mixing javascript and server-side code so let me know if there's a more elegant solution.
        var ids = "${attrs.id}"
        var dataType = "${attrs.dataType}";
        var templateString = "${attrs.template}"

        // Make ajax call to pull data and 
        $.ajax({
            dataType: "json",
            timeout: ${attrs.timeout},
            url: "${attrs.url}",
            success: function (data) {
                $.each(ids, function(id) { 
                    var templateString = 
                    var value = data.get(id)
                    switch(dataType) { 
                        case "decimal": 
                            value = value.toFixed(1)
                            break;
                        ...
                    }
                    // FIXME psuedocode - not sure how to do this but assume it could be done with plain
                    // javascript OR with a simple template engine
                    var renderedHtml = templateString.doInterpolation(value);
                    $(id).html(renderedHtml);
                });
            },
            error: function(error) { 
                // implement error handling
            }
          });
        });
    </g:javascript>


You could also have the taglib support a list of data points, but I think that would be overkill given that we're only making a few extra ajax calls per page. Perhaps if we were talking about multiple rows of data points we might have to do something like this. But we already have solutions to that problem. Including the "don't use JQuery" solution.

    <g:remoteRequest 
        url="${request.contextPath}/json/getForecastingData/${productInstance?.id}"
        id="['#onHandMonths','#monthlyDemand'] 
        value=['onHandMonths','monthlyDemand'] 
        dataType="decimal"
        template="['${localOnHandMonths} months', '${localMonthlyDemand} per month'] />

Reply:
I have changed commit and added timeout from Config.